### PR TITLE
Add offline node purge timeout in cluster config

### LINKED ISSUE #1484: Add offline node purge timeout in cluster config
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is.
Helix plans to support the purge of offline nodes that have passed the timeout defined by users. As a prerequisite, we need to provide a simple field in cluster config for users to define the default timeout. 

**Describe the solution you'd like**
Add one more field in cluster config for offline nodes timeout to be purged.

**Additional context**

### PR DESCRIPTION
**Pull request by zhangmeng916**
### Issues

- [X] My PR addresses the following Helix issues and references them in the PR description:

Fixed #1484 

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

Helix plans to support the purge of offline nodes that have passed the timeout defined by users. As a prerequisite, we need to provide a simple field in cluster config for users to define the default timeout. This PR add one more field in cluster config for offline nodes timeout to be purged.

### Tests

- [X] The following tests are written for this issue:

TestClusterConfig

- [X] The following is the result of the "mvn test" command on the appropriate module:
helix-core:
[ERROR] Failures:
[ERROR]   TestRoutingTableProviderPeriodicRefresh.testPeriodicRefresh:214 expected:<4> but was:<3>
[ERROR]   TestDeleteJobFromJobQueue.testForceDeleteJobFromJobQueue:73 » Helix Failed to ...
[INFO]
[ERROR] Tests run: 1236, Failures: 2, Errors: 0, Skipped: 1
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:22 h
[INFO] Finished at: 2020-10-27T13:12:36-07:00

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

**Review by dasahcc — APPROVED** [review]
Overall, LGTM

File: helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java [hunk 1]
```diff
@@ -937,6 +940,27 @@ public long getOfflineNodeTimeOutForMaintenanceMode() {
...
+   * the node will be dropped.
+   * @param timeOut timeout window in milliseconds.
+   */
+  public void setOfflineNodeTimeOutForPurge(long timeOut) {
+    _record.setLongField(ClusterConfigProperty.OFFLINE_NODE_TIME_OUT_FOR_PURGE.name(),
+        timeOut);
+  }
+
+  /**
+   * Get the default time out window for offline nodes to be purged. I
```
Comment (kaisun2000) [inline helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java:955]: nit, additional "I" at end of this line.

File: helix-core/src/test/java/org/apache/helix/model/TestClusterConfig.java [hunk 2]
```diff
@@ -320,6 +320,29 @@ public void testSetOfflineNodeTimeOutForMaintenanceMode() {
...
+
+    testConfig.getRecord()
+        .setLongField(ClusterConfig.ClusterConfigProperty.OFFLINE_NODE_TIME_OUT_FOR_PURGE
+                .name(),
+            10000L);
+    Assert.assertEquals(testConfig.getOfflineNodeTimeOutForPurge(), 10000L);
+  }
+
+  @Test
+  public void testSetOfflineNodeTimeOutForPurge() {
```
Comment (kaisun2000) [inline helix-core/src/test/java/org/apache/helix/model/TestClusterConfig.java:337]: This test really is not necessary. All the logic seems tested in testGetOfflineNodeTimeOutForPurge() anyway.

**Review by kaisun2000 — APPROVED** [review]

File: helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java [hunk 3]
```diff
@@ -128,7 +128,9 @@
     // offline for more than this specified time period, it's treated as offline for the rest of
     // the maintenance mode's duration even when it comes online.
     // The unit is milliseconds.
-    OFFLINE_NODE_TIME_OUT_FOR_MAINTENANCE_MODE
+    OFFLINE_NODE_TIME_OUT_FOR_MAINTENANCE_MODE,
+
+    OFFLINE_NODE_TIME_OUT_FOR_PURGE
```
Comment (jiajunwang) [inline helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java:133]: nit, comment to explain the option please

File: helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java [hunk 4]
```diff
@@ -158,6 +160,7 @@
   public final static boolean DEFAULT_GLOBAL_REBALANCE_ASYNC_MODE_ENABLED = true;
   private static final int GLOBAL_TARGET_TASK_THREAD_POOL_SIZE_NOT_SET = -1;
   private static final long OFFLINE_NODE_TIME_OUT_FOR_MAINTENANCE_MODE_NOT_SET = -1;
+  private static final long OFFLINE_NODE_TIME_OUT_FOR_PURGE_NOT_SET = -1;
```
Comment (jiajunwang) [inline helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java:163]: not necessarily to be changed here, but maybe we want to have one single static var VALUE_NOT_SET = -1.

File: helix-core/src/test/java/org/apache/helix/model/TestClusterConfig.java (hunk 2 above)
Comment (zhangmeng916) [reply helix-core/src/test/java/org/apache/helix/model/TestClusterConfig.java:337]: Tiny difference in how they set the field. But I agree they're doing the same thing.

**Comment by zhangmeng916** [comment]
This PR is ready to merge. Approved by @dasahcc and @kaisun2000 
Final commit message: Add offline node purge timeout in cluster config

**Pull request merged**